back-end/www/models/model_operations/user_operations.py

```diff
@@ -129,3 +129,61 @@ def remove_user(user_id):
 
     db.session.delete(user)
     db.session.commit()
+
+
+def get_user_count():
+    """
+    Get total user count.
+
+    Returns
+    -------
+    count : int
+        Number of total users.
+    """
+    count = User.query.count()
+
+    return count
+
+
+def get_user_done_location_count(client_id):
```

Comment:
I have a feeling that this function is going to take a lot of time to run, since you are trying to access the database for so many times in this function. Think about if we can simplify it and make it run faster.

Here is something that I would suggest:
1. Join the user.answers (after filtering out the gold standards) and the location table, using location_id
2. Use something like user.answers.location to get the locations that the user labeled
3. Count the number of locations with timestamps (not None)

Reply:
But let's keep this as low priority. For now, if the function is working, just put a TODO flag to indicate that you will revise the function later. So that you can move on and work on the more important things.

Reply:
I re-write according to the answer process clarification. For individual done count, because the answer gets into the DB only when the user passes the quality check, all the answers for the user in the DB are considered to be "done", right?

Reply:
Just want to double check that if `get_user_done_location_count` means how many locations one single user (`client_ID`) has answered?

As for the criteria for answers to be recorded, the latest decision on Dec. 8 is that all completed answers will go into DB no matter the answer set passes the gold standard test or not. The rationale behind that decision is 
1. The main goal is civic participation instead of collecting high quality data only. The data of how many users and answers collected can be leveraged in the advocacy, e.g. at least 10k citizens care about this issue to mark 20k factory locations in 10 days This alone can already press the government to respond and be on media. However, that doesn't mean we don't care the quality of crowdsourcing data. We still need indicators for us to monitor and increase the data quality. Better data bring us more bargaining power in the issue and for future academic research. I assume that we can filter the data passing golden standard afterwards. The results of SpotDiff will not be realtime displayed on disfactory.tw. The data will only publish after careful examination.
2. We need data to know the passing rate of gold standard since the diff process is still not easy enough to ensure, so I don't want to drop the unsuccessful data. I assume that recording failed answer sets won't create too much burden on our server. Correct me if my assumption is wrong.

Reply:
Today we add `gold_standard_status` in data model to mark whether the answer is from gold standard passing group 

(0: is gold standard, 1: pass, 2: fail)


https://docs.google.com/presentation/d/1hyak0PdXA3CpxSC82T7ahmp0AbqjEJS2c3AcnhURpnw/edit#slide=id.gf5e3502c4a_2_0